test/Dialect/MSFT/partition.mlir

```diff
@@ -13,14 +21,21 @@ msft.module @top {} (%clk : i1) -> (out1: i2, out2: i2) {
 
 msft.module.extern @Extern (%foo_a: i2) -> (foo_x: i2)
 
-msft.module @B {} (%clk : i1) -> (x: i2)  {
+msft.module @B {} (%clk : i1) -> (x: i2, y: i2)  {
   %c1 = hw.constant 1 : i2
-  %0 = msft.instance @unit1 @Extern(%c1) { targetDesignPartition = @top::@part1 }: (i2) -> (i2)
+  %0 = msft.instance @unit1 @Extern(%c1) { targetDesignPartition = @top::@part1, circt.globalRef = [#hw.globalNameRef<@ref1>], inner_sym = "unit1" }: (i2) -> (i2)
   %1 = seq.compreg %0, %clk { targetDesignPartition = @top::@part1 } : i2
 
   %2 = msft.instance @unit2 @Extern(%1) { targetDesignPartition = @top::@part1 }: (i2) -> (i2)
 
-  msft.output %2: i2
+  %3 = msft.instance @c @C(%2) { targetDesignPartition = @top::@part1, circt.globalRef = [#hw.globalNameRef<@ref2>], inner_sym = "c" }: (i2) -> (i2)
+
+  msft.output %2, %3: i2, i2
+}
+
+msft.module @C {} (%in : i2) -> (out: i2)  {
```

Comment:
I was gonna ask for a test with a hierarchy depth >2. Glad I saw this before asking 😜.